server/claim/claim.service.ts

```diff
@@ -46,80 +54,78 @@ export class ClaimService {
-                    await this.sourceService.create({
-                        link: claim.sources[i],
-                        targetId: newClaim.id,
-                        targetModel: "Claim",
-                    });
-                }
-            } catch (e) {
-                this.logger.error(e);
-                throw e;
-            }
+        const newClaimRevision = await this.claimRevisionService.create(newClaim._id, claim)
+        newClaim.latestRevision = newClaimRevision._id
+        newClaim.slug = newClaimRevision.slug
+        newClaim.save();
+
+        return {
+            ...newClaim.toObject(),
+            ...newClaimRevision.toObject()
         }
-        return newClaim.save();
     }
 
-    async update(claimId, claimBody) {
-        // eslint-disable-next-line no-useless-catch
-        try {
-            const claim = await this.getById(claimId);
-            claimBody.content = this.parserService.parse(claimBody.content);
-            const newClaim = Object.assign(claim, claimBody);
-            const claimUpdate = await this.ClaimModel.findByIdAndUpdate(
-                claimId,
-                newClaim,
-                this.optionsToUpdate
-            );
-            return claimUpdate;
-        } catch (error) {
-            // TODO: log to service-runner
-            throw error;
-        }
+    async update(claimId, claimRevisionUpdate) {
+        const claim = await this._getClaim({ _id: claimId }, false);
+        const latestRevision = claim.toObject().latestRevision
+        delete latestRevision._id
+        const newClaimRevision =
+            await this.claimRevisionService.create(claim._id, {
+                ...latestRevision,
+                ...claimRevisionUpdate
+            })
+        claim.latestRevision = newClaimRevision._id
+        claim.slug = newClaimRevision.slug
+        claim.save()
+        return newClaimRevision;
     }
 
     delete(claimId) {
+        // TODO: use soft-delete instead
+        // this means that when deleting it should actually update
+        // the deleted field with the boolean value 'true'
         return this.ClaimModel.findByIdAndRemove(claimId);
     }
 
+    // TODO: add optional revisionId that will fetch a specifc revision that matches
     async getById(claimId) {
-        const claim = await this.ClaimModel.findById(claimId)
-            .populate("personality", "_id name")
-            .populate("sources", "_id link classification");
-        if (!claim) {
-            // TODO: handle 404 for claim not found
-            return {};
-        }
-
-        return this.postProcess(claim.toObject());
+        return this._getClaim({_id: claimId})
     }
 
     async getByPersonalityIdAndClaimSlug(personalityId, claimSlug) {
+        return this._getClaim({personality: personalityId, slug: claimSlug})
+    }
+
+    private async _getClaim(match: ClaimMatchParameters, postprocess = true) {
```

Comment:
@pepermao I meant that we have created this new function that retrieves Claim with a revision, and execute the postProcess function.

In the comment above we should use _getClaim and refactor the functionality.

Reply:
But let's do this in a another PR because it's going to be a little tricky.